src/features/home/pages/Home.native.test.tsx

```diff
@@ -63,59 +62,76 @@ describe('Home component', () => {
   mockUseNetInfoContext.mockReturnValue({ isConnected: true })
 
   afterEach(jest.clearAllMocks)
-  beforeEach(() => {
-    mockUserProfileInfo = {
-      email: 'email@domain.ext',
-      firstName: 'Jean',
-      isBeneficiary: true,
-      depositExpirationDate: '2023-02-16T17:16:04.735235',
-      domainsCredit: {
-        all: { initial: 50000, remaining: 49600 },
-        physical: { initial: 20000, remaining: 0 },
-        digital: { initial: 20000, remaining: 19600 },
-      },
-    }
-  })
 
   it('should render skeleton when there are no modules to display', () => {
     mockUseHomepageData.mockReturnValueOnce({
       modules: [],
       homeEntryId: 'fake-entry-id',
       thematicHeader: { title: 'HeaderTitle', subtitle: 'HeaderSubtitle' },
     })
+    mockUseAuthContext.mockReturnValueOnce({ isLoggedIn: true, user: beneficiaryUser })
+    mockUseAvailableCredit.mockReturnValueOnce({ amount: 49600, isExpired: false })
 
     const home = render(<Home />)
     expect(home).toMatchSnapshot()
   })
 
   it('should render correctly without login modal', async () => {
-    mockUserProfileInfo = undefined
+    mockUseAuthContext.mockReturnValueOnce({
+      isLoggedIn: false,
+    })
     env.FEATURE_FLIPPING_ONLY_VISIBLE_ON_TESTING = false
     const { toJSON } = render(<Home />)
     expect(toJSON()).toMatchSnapshot()
   })
 
   it('should have a welcome message', async () => {
-    mockUserProfileInfo = undefined
+    mockUseAuthContext.mockReturnValueOnce({
+      isLoggedIn: false,
+    })
     const { getByText } = render(<Home />)
     await waitFor(() => getByText('Bienvenue\u00a0!'))
   })
 
   it('should have a personalized welcome message when user is logged in', async () => {
+    // eslint-disable-next-line local-rules/independent-mocks
+    mockUseAuthContext.mockReturnValue({
+      isLoggedIn: true,
+      user: beneficiaryUser,
+      isUserLoading: false,
+    })
```

Comment:
Attention, un `mockReturnValue` sauvage attaque !
Est-ce que tu peux justifier pourquoi, et dans tous les cas, reset le mock à la fin pour être comme à l'état initial ?

Reply:
Même remarque pour les 4 autres itérations dans ce fichier